src/components/modal/index.js

```diff
@@ -33,7 +33,11 @@ export default class Modal extends BaseComponent {
     /**
      * the background color of the overlay
      */
-    overlayBackgroundColor: PropTypes.string
+    overlayBackgroundColor: PropTypes.string,
+    /**
+     * The modal's end-to-end test identifier
+     */
+    testID: PropTypes.string
```

Comment:
Just declaring the prop doesn't really do anything. 
You will actually need to use the testID inside the component. (See the link for I added in the previous comment to stack overflow)

React Native's Modal component doesn't support testID internally, so we will need to pass it ourselves to the wrapper view inside the Modal.